hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -113,6 +124,24 @@ public void initialize(String inputName, Map<String, String> properties) {
     this.clients = new CachedClientPool(conf, properties);
   }
 
+  @Override
+  public TableBuilder buildTable(TableIdentifier identifier, Schema schema) {
+    if (viewExists(identifier)) {
```

Comment:
This might not be the right place to do this validation. But other catalogs are doing it in different way like [JDBC](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java#L820)  doing the first check . [Nessie](https://github.com/apache/iceberg/blob/main/nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java#L132)  during commit with exceptions, [InMemory](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java#L404)  during commit too. 

In case of Hive, we should do before `doRefresh` Since we are the tableType check [Here](https://github.com/apache/iceberg/blob/main/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L147).  

@szehon-ho  Please share your thoughts.

Reply:
yes this is the wrong place to do this validation. You're already doing this validation in https://github.com/apache/iceberg/pull/9852/files#diff-5ecfc223b311b523a12be7482b6235318fdd7535c371f899daf7e09eacddad77R322